lib/grape/validations/validators/coerce.rb

```diff
@@ -56,7 +56,14 @@ def coerce_value(type, val)
         return val || Set.new if type == Set
         return val || {}      if type == Hash
 
-        converter = Virtus::Attribute.build(type)
+        # To support custom types that Virtus can't easily coerce, pass in an
+        # explicit coercer. Custom types must implement a `parse` class method.
+        converter_options = {}
+        if ParameterTypes.custom_type?(type)
+          converter_options[:coercer] = type.method(:parse)
+        end
```

Comment:
I'd have to dig :) I think it should tell you to do `converter_options[:coercer] = type.method(:parse) if ...`. It's really nbd.